### PR TITLE
spanner: code samples for query options

### PR DESCRIPTION
**Pull request by hengfengli**

### CODE REVIEW

**Comment by hengfengli** [comment]
@skuruppu I have updated the code samples and test locally. It works fine. 

I run the following command to update the spanner to v1.3.0: 

```bash
go get -v -u cloud.google.com/go/spanner
```

But somehow, it updates other modules as well. Do you think it is okay to merge it?

File: spanner/spanner_snippets/snippet.go [hunk 1]
```diff
@@ -88,6 +90,11 @@ var (
 		"querywithint":                queryWithInt,
 		"querywithstring":             queryWithString,
 		"querywithtimestampparameter": queryWithTimestampParameter,
+		"querywithqueryoptions":       queryWithQueryOptions,
+	}
+
+	newClientCommands = map[string]newClientCommand{
```
Comment (tbpg) [inline spanner/spanner_snippets/snippet.go]: s/newClientCommands/newClientCommand/g

File: spanner/spanner_snippets/snippet.go [hunk 2]
```diff
@@ -1819,6 +1853,41 @@ func queryWithHistory(ctx context.Context, w io.Writer, client *spanner.Client)
 	}
 }
 
+// [START spanner_create_client_with_query_options]
+
+func createClientWithQueryOptions(ctx context.Context, w io.Writer, database string) error {
+	queryOptions := spanner.QueryOptions{Options: &sppb.ExecuteSqlRequest_QueryOptions{OptimizerVersion: "1"}}
+	client, err := spanner.NewClientWithConfig(
+		ctx, database, spanner.ClientConfig{QueryOptions: queryOptions},
+	)
```
Comment (tbpg) [inline spanner/spanner_snippets/snippet.go]: Don't think the newline is needed?

https://github.com/golang/go/wiki/CodeReviewComments#line-length

File: spanner/spanner_snippets/snippet.go (hunk 1 above)
Comment (tbpg) [reply spanner/spanner_snippets/snippet.go]: Or with the s, but needs to be consistent. :)

Comment (hengfengli) [reply spanner/spanner_snippets/snippet.go]: `newClientCommands` is a collection (mapping from string to newClientCommand). If you look at other declarations in this file, this is consistent. 

https://github.com/GoogleCloudPlatform/golang-samples/blob/1bd6ff8a213e97ad19565b4a636a4a82405c2cc5/spanner/spanner_snippets/snippet.go#L43

https://github.com/GoogleCloudPlatform/golang-samples/blob/1bd6ff8a213e97ad19565b4a636a4a82405c2cc5/spanner/spanner_snippets/snippet.go#L100

File: spanner/spanner_snippets/snippet.go (hunk 2 above)
Comment (hengfengli) [reply spanner/spanner_snippets/snippet.go]: As far as I can see, there is not line length limit in the getting started guide for Go (https://cloud.google.com/spanner/docs/getting-started/go). But this is a good point. Fixed.

**Comment by skuruppu** [comment]
@hengfengli and @tbpg it would be really great to get this set of samples submitted because the docs team is blocked on completing the documentation.

**Comment by hengfengli** [comment]
> I merged the latest changes, but the diff looks a little weird now. Sorry about that.
> 
> Looks like there is a build failure, though?
> 
> ```
> ##[error]spanner/spanner_snippets/snippet.go:1615:18: undefined: "cloud.google.com/go/spanner".QueryOptions
> ##[error]spanner/spanner_snippets/snippet.go:1618:25: client.Single().QueryWithOptions undefined (type *"cloud.google.com/go/spanner".ReadOnlyTransaction has no field or method QueryWithOptions)
> ##[error]spanner/spanner_snippets/snippet.go:1863:18: undefined: "cloud.google.com/go/spanner".QueryOptions
> ##[error]spanner/spanner_snippets/snippet.go:1867:39: unknown field 'QueryOptions' in struct literal of type "cloud.google.com/go/spanner".ClientConfig
> ```
> 
> See the "Build and Lint" check.

Thanks a lot for the fix. 😊This is because the spanner is still in an old version v1.1.0. I will upgrade it to the latest version v1.4.0.

**Comment by hengfengli** [comment]
@tbpg I will rollback your changes and rebase the branch. This will look a bit better.

**Comment by tbpg** [comment]
Failure is unrelated flake. Merging.

**Pull request merged**